components/cosmetic_filters/renderer/cosmetic_filters_js_handler.cc

```diff
@@ -116,6 +116,51 @@ const char kHideSelectorsInjectScript[] =
           };
         })();)";
 
+const char kRemovalsInjectScript[] =
+    R"((function() {
+          const selectorsToRemove = %s;
+          const classesToRemoveBySelector = %s;
+          const attributesToRemoveBySelector = %s;
+          const onMutation = (mutationList) => {
+            if (mutationList.find(
+                mutation => (
+                  mutation.addedNodes.length !== 0 && mutation.addedNodes.values().find(node => node.nodeType == 1)
+                )
+              )
+            ) {
+              execute();
+            }
+          };
+          const execute = () => {
+              selectorsToRemove.forEach(
+                selector => document.querySelectorAll(selector).forEach(elem => elem.remove())
+              );
+              for (const selector in classesToRemoveBySelector) {
+                const classesToRemove = classesToRemoveBySelector[selector];
+                document.querySelectorAll(selector).forEach(elem => {
+                  const classesToActuallyRemove = classesToRemove.filter(elem.classList.contains.bind(elem.classList));
+                  if (classesToActuallyRemove.length) {
+                    elem.classList.remove(classesToActuallyRemove);
```

Comment:
Could we just call `elem.classList.remove(classesToRemove)` ?
I believe it's faster and more clear. And should also match to the specs.

Reply:
Perhaps. The reason I didn't is that `elem.classList.remove("non-existent-class")` will trigger the MutationObserver again if the observer is observing attribute changes, even though no mutation is actually performed.

Reply:
Really?
It sounds like a blink bug to me. Maybe check the specs and report to crbug.com?

Reply:
The specs doesn't anything about a possibility to skip serializing `classList` if it wasn't changed.
https://dom.spec.whatwg.org/#dom-domtokenlist-remove

Reply:
BTW, what is the complexity of such filtering? `O(classesToRemove.lenght * elem.classList.lenght)`?

Reply:
I think `classesToRemove` will only ever be of length 1. Is that correct @antonok-edm that the `Vec`s in `remove_attrs: HashMap<String, Vec<String>>` and `remove_classes` can only be of length one? I can't find the syntax to specify more than one.

Reply:
Ah I'm wrong, a rule can only have one class to remove, but you can add multiple rules for the same selector and then the Vec has multiple elements.